File: examples/c++/a110x.cxx
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2014 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>
#include <signal.h>

#include "a110x.hpp"
#include "upm_utilities.h"

using namespace std;

int shouldRun = true;

void
sig_handler(int signo)
{
    if (signo == SIGINT)
        shouldRun = false;
}

int
main()
{
    signal(SIGINT, sig_handler);

    //! [Interesting]
    // Instantiate an A110X sensor on digital pin D2
    upm::A110X hall(2);

    // check every second for the presence of a magnetic field (south
    // polarity)
    while (shouldRun) {
        bool val = hall.magnetDetected();
        if (val)
            cout << "Magnet (south polarity) detected." << endl;
        else
            cout << "No magnet detected." << endl;

        upm_delay(1);
    }
    //! [Interesting]

    cout << "Exiting..." << endl;

    return 0;
}

File: examples/javascript/a110x.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2014 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var hallEffectSensor = require('jsupm_a110x');

// Instantiate a Hall Effect magnet sensor on digital pin D2
var myHallEffectSensor = new hallEffectSensor.A110X(2);

// Check every second for the presence of a magnetic field (south polarity)
setInterval(function()
{
	if (myHallEffectSensor.magnetDetected())
		console.log("Magnet (south polarity) detected.");
	else
		console.log("No magnet detected.");
}, 1000);

// Print message when exiting
process.on('SIGINT', function()
{
	console.log("Exiting...");
	process.exit(0);
});
